### PR TITLE
Icu 6705 add stylelint to rose

### PR DESCRIPTION
**Pull request by ZedLi**
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-6705)

## Description
Added stylelint to rose. Disabled a few rules with comments explaining why.

### CODE REVIEW

File: addons/rose/.stylelintrc.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+module.exports = {
+  extends: ['stylelint-config-standard-scss', 'stylelint-config-prettier-scss'],
```
Comment (ZedLi) [inline addons/rose/.stylelintrc.js:4]: Used `stylelint-config-standard-scss` over `stylelint-config-sass-guidelines` which we're using in desktop. This one seems to be better supported now and is recommended by the official docs.

File: addons/rose/app/styles/rose/components/form/_radio-card.scss [hunk 2]
```diff
@@ -180,18 +196,20 @@ $radio-svg: (
 
   &.error {
     .rose-form-radio-field {
-      &:not(:disabled):not(:focus) {
-        &:not(:disabled):not(:focus) {
+      &:not(:disabled, :focus) {
+        &:not(:disabled, :focus) {
```
Comment (randallmorey) [inline addons/rose/app/styles/rose/components/form/_radio-card.scss:200]: Are these duplicated?  Not sure why they were in the first place, but probably unnecessary?

File: addons/rose/package.json [hunk 3]
```diff
@@ -96,12 +97,16 @@
     "jest": "^27.4.7",
     "loader.js": "^4.7.0",
     "npm-run-all": "^4.1.5",
+    "postcss": "^8.4.18",
```
Comment (randallmorey) [inline addons/rose/package.json:100]: Why is PostCSS added?

**Review by randallmorey — CHANGES_REQUESTED** [review]

File: addons/rose/package.json (hunk 3 above)
Comment (calcaide) [reply addons/rose/package.json:100]: Curious about this too.

Comment (ZedLi) [reply addons/rose/package.json:100]: Forgot to comment about this, stylelint 14 [does not support any non-css](https://stylelint.io/migration-guide/to-14#syntax-option-and-automatic-inferral-of-syntax) syntax out of the box anymore. 

Both [stylelint-config-standard-scss](https://github.com/stylelint-scss/stylelint-config-standard-scss/blob/main/package.json#L64) and [stylelint-config-sass-guidelines](https://github.com/bjankord/stylelint-config-sass-guidelines/blob/main/package.json#L38) have a peer dependency on `postcss` so they won't work without it.

File: addons/rose/app/styles/rose/components/form/_radio-card.scss (hunk 2 above)
Comment (calcaide) [reply addons/rose/app/styles/rose/components/form/_radio-card.scss:200]: Indeed, seems duplicate and unneccesary states. I tested within storybook and I couldn't reproduce to show this specific not diabled and not focus.

File: addons/rose/package.json (hunk 3 above)
Comment (calcaide) [reply addons/rose/package.json:100]: Ahh! that makes all the sense then 😉

**Review by calcaide — APPROVED** [review]

File: addons/rose/app/styles/rose/components/form/_radio-card.scss (hunk 2 above)
Comment (ZedLi) [reply addons/rose/app/styles/rose/components/form/_radio-card.scss]: Looks like it, removed it

**Review by randallmorey — APPROVED** [review]

**Review by cameronperera — APPROVED** [review]

**Pull request merged**